content/ch-quantum-hardware/Jaynes-Cummings-model.ipynb

```diff
@@ -0,0 +1,764 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "%matplotlib inline\n",
+    "# Importing standard Qiskit libraries and configuring account\n",
+    "from qiskit import QuantumCircuit, execute, Aer, IBMQ\n",
+    "from qiskit.compiler import transpile, assemble\n",
+    "from qiskit.tools.jupyter import *\n",
+    "from qiskit.visualization import *\n",
+    "from qiskit.visualization import SchedStyle\n",
+    "# Loading your IBM Q account(s)\n",
+    "provider = IBMQ.load_account()\n",
+    "IBMQ.get_provider(hub='ibm-q', group='open', project='main')\n",
+    "backend = provider.get_backend('ibmq_armonk')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We then extract the default backend configuration and settings for the selected chip"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "/opt/conda/lib/python3.7/site-packages/qiskit/providers/ibmq/ibmqfactory.py:192: UserWarning: Timestamps in IBMQ backend properties, jobs, and job results are all now in local time instead of UTC.\n",
+      "  warnings.warn('Timestamps in IBMQ backend properties, jobs, and job results '\n"
+     ]
+    }
+   ],
+   "source": [
+    "backend_config = backend.configuration()\n",
+    "backend_defaults = backend.defaults()\n",
+    "backend.configuration().parametric_pulses = []  # Will allow us to send a larger waveform for our experiments\n",
+    "style = SchedStyle(figsize=(3, 2), title_font_size=10, axis_font_size=8) # style for displaying the pulse sequence"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Next we define some helper functions that we will use for fitting and interpreting our data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from scipy.optimize import curve_fit\n",
+    "from scipy.signal import savgol_filter\n",
+    "\n",
+    "# samples need to be multiples of 16 to accommodate the hardware limitations\n",
+    "def get_closest_multiple_of_16(num):\n",
+    "    return int(num + 8 ) - (int(num + 8 ) % 16)\n",
+    "\n",
+    "# process the reflective measurement results\n",
+    "# in a reflective measurement the data is encoded in the phase of the output signal\n",
+    "def process_reflective_measurement(freqs,values):\n",
+    "    phase_grad = np.gradient(savgol_filter(np.unwrap(np.angle(values)),3,2),freqs)\n",
+    "    return (phase_grad-min(phase_grad))/(max(phase_grad)-min(phase_grad)) - 1\n",
+    "\n",
+    "# lorentzian function\n",
+    "def lorentzian(f, f0, k, a, offs):\n",
+    "    return -a*k/(2*np.pi)/((k/2)**2+(f-f0)**2)+offs\n",
+    "\n",
+    "# fit a lorentizan function\n",
```

Comment:
This comment is not especially helpful as the reader could probably guess this from the function name. I would go for a description of its behaviour, e.g. 
>`fit_lorentzian` takes two arrays that contain the x & y values of each data point respectively. It returns an estimate for the parameters that best reproduce this data. Output is... (explain what `popt` and `pcov` are).